lib/TerminalDisplay.cpp

```diff
@@ -1230,15 +1260,19 @@ void TerminalDisplay::focusInEvent(QFocusEvent*)
 void TerminalDisplay::paintEvent( QPaintEvent* pe )
 {
   QPainter paint(this);
-
-  foreach (const QRect &rect, (pe->region() & contentsRect()).rects())
-  {
-    drawBackground(paint,rect,palette().background().color(),
-                    true /* use opacity setting */);
-    drawContents(paint, rect);
+  
+  if(_drawTextTestFlag) {
+      calDrawTextAdditionHeight(paint);      
+  } else {
```

Comment:
Doesn't this else block prevent the paintEvent from painting what it should have if the draw padding needed an update?  If so, remove the else block.